src/transport_picoquic.cpp

```diff
@@ -0,0 +1,828 @@
+}
+
+void PicoQuicTransport::setStatus(TransportStatus status) {
+  transportStatus = status;
+}
+
+StreamId PicoQuicTransport::createStream(
+  const TransportContextId& context_id,
+  bool use_reliable_transport)
+{
+  logger.log(LogLevel::debug, "App requests to create new stream");
+
+  const auto &iter = active_streams.find(context_id);
+  if (iter == active_streams.end()) {
+    logger.log(LogLevel::warn, "Invalid context id, cannot create stream");
+    return 0;
+  }
+
+  const auto &cnx_stream_iter = iter->second.find(0);
+  if (cnx_stream_iter == iter->second.end()) {
+    logger.log(LogLevel::warn,
+               "Missing primary connection zero stream, cannot create streams");
+    return 0;
+  }
+
+  if (isServerMode) // Server does not create streams, clients must create these
+    return 0;
+  else if (not use_reliable_transport)
+    return 0;
+
+  // Create stream will add stream to active_streams
+  PicoQuicTransport::StreamContext * stream_cnx = createStreamContext(
+      cnx_stream_iter->second.cnx,
+      next_stream_id);
+
+  TransportContextId tcid = context_id;
+  StreamId stream_id = next_stream_id;
+
+  cbNotifyQueue.push([&] () {
+    delegate.on_new_stream(tcid, stream_id);
+  });
+
+  // Set next stream ID
+  next_stream_id += 4; // Increment by 4, stream type is first 2 bits
+
+  return stream_cnx->stream_id;
+}
+
+TransportContextId
+PicoQuicTransport::start()
+{
+  TransportContextId cid = 0;
+  uint64_t current_time = picoquic_current_time();
+  debug_set_stream(stdout);  // Enable picoquic debug
+
+  (void)picoquic_config_set_option(&config,
+                                   picoquic_option_ALPN, QUICR_ALPN);
+  (void)picoquic_config_set_option(&config,
+                                   picoquic_option_MAX_CONNECTIONS, "100");
+  quic_ctx = picoquic_create_and_configure(&config,
+                                           pq_event_cb,
+                                           this,
+                                           current_time,
+                                           NULL);
+
+  if (quic_ctx == NULL) {
+    logger.log(LogLevel::fatal, "Unable to create picoquic context, check certificate and key filenames");
+    throw PicoQuicException("Unable to create picoquic context");
+  }
+
+  /*
+   * TODO doc: Apparently need to set some value to send datagrams. If not set,
+   *    max datagram size is zero, preventing sending of datagrams. Setting this
+   *    also triggers PMTUD to run. This value will be the initial value.
+   */
+  picoquic_init_transport_parameters(&local_tp_options, 1);
+  local_tp_options.max_datagram_frame_size = 1400;
+//  local_tp_options.max_ack_delay = 1000000;
+//  local_tp_options.min_ack_delay = 20000;
+
+  picoquic_set_default_tp(quic_ctx, &local_tp_options);
+
+  cbNotifyThread = std::thread(&PicoQuicTransport::cbNotifier, this);
+
+  if (isServerMode) {
+    picoQuicThread = std::thread(&PicoQuicTransport::server, this);
+    return 0;
+
+  } else {
```

Comment:
Since we return in the if statement, else is redundant.

Reply:
IMO, it's easier to read with the else.. ```cid``` is initialized to ZERO, so it should be fine to remove the ```return 0```.